// @solana/spl-token/lib/cjs/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NATIVE_MINT_2022 = exports.NATIVE_MINT = exports.ASSOCIATED_TOKEN_PROGRAM_ID = exports.TOKEN_2022_PROGRAM_ID = exports.TOKEN_PROGRAM_ID = void 0;
exports.programSupportsExtensions = programSupportsExtensions;
const web3_js_1 = require("@solana/web3.js");
/** Address of the SPL Token program */
exports.TOKEN_PROGRAM_ID = new web3_js_1.PublicKey('6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b');
/** Address of the SPL Token 2022 program */
exports.TOKEN_2022_PROGRAM_ID = new web3_js_1.PublicKey('6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b');
/** Address of the SPL Associated Token Account program */
exports.ASSOCIATED_TOKEN_PROGRAM_ID = new web3_js_1.PublicKey('6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b');
/** Address of the special mint for wrapped native SOL in spl-token */
exports.NATIVE_MINT = new web3_js_1.PublicKey('6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b');
/** Address of the special mint for wrapped native SOL in spl-token-2022 */
exports.NATIVE_MINT_2022 = new web3_js_1.PublicKey('6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b');
/** Check that the token program provided is not `Tokenkeg...`, useful when using extensions */
function programSupportsExtensions(programId) {
    if (programId.equals(exports.TOKEN_PROGRAM_ID)) {
        return false;
    }
    else {
        return true;
    }
}
