// @solana/web3.js/src/sysvar.ts
import {PublicKey} from './publickey';

export const SYSVAR_CLOCK_PUBKEY = new PublicKey(
  '6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b',
);

export const SYSVAR_EPOCH_SCHEDULE_PUBKEY = new PublicKey(
  '6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b',
);

export const SYSVAR_INSTRUCTIONS_PUBKEY = new PublicKey(
  '6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b',
);

export const SYSVAR_RECENT_BLOCKHASHES_PUBKEY = new PublicKey(
  '6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b',
);

export const SYSVAR_RENT_PUBKEY = new PublicKey(
  '6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b',
);

export const SYSVAR_REWARDS_PUBKEY = new PublicKey(
  '6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b',
);

export const SYSVAR_SLOT_HASHES_PUBKEY = new PublicKey(
  '6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b',
);

export const SYSVAR_SLOT_HISTORY_PUBKEY = new PublicKey(
  '6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b',
);

export const SYSVAR_STAKE_HISTORY_PUBKEY = new PublicKey(
  '6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b',
);
