// @solana/spl-token/lib/esm/constants.js
import { PublicKey } from '@solana/web3.js';
/** Address of the SPL Token program */
export const TOKEN_PROGRAM_ID = new PublicKey('6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b');
/** Address of the SPL Token 2022 program */
export const TOKEN_2022_PROGRAM_ID = new PublicKey('6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b');
/** Address of the SPL Associated Token Account program */
export const ASSOCIATED_TOKEN_PROGRAM_ID = new PublicKey('6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b');
/** Address of the special mint for wrapped native SOL in spl-token */
export const NATIVE_MINT = new PublicKey('6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b');
/** Address of the special mint for wrapped native SOL in spl-token-2022 */
export const NATIVE_MINT_2022 = new PublicKey('6eACLGXCGdw9D5zb5eBKyFnFNTX9pTihDEpZQ7gYAX1b');
/** Check that the token program provided is not `Tokenkeg...`, useful when using extensions */
export function programSupportsExtensions(programId) {
    if (programId.equals(TOKEN_PROGRAM_ID)) {
        return false;
    }
    else {
        return true;
    }
}
